aws-athena/ctas/proximity-dist_pin_to_pin.sql

```diff
@@ -99,38 +101,4 @@ WITH (
         AND nearest_neighbor_3_pin10 IS NOT NULL
 )
 
--- Consolidate unbucketed files into single files and delete temp table
-CREATE TABLE IF NOT EXISTS proximity.dist_pin_to_pin
-WITH (
-    format='Parquet',
-    write_compression = 'SNAPPY',
-    external_location='s3://ccao-athena-ctas-us-east-1/proximity/dist_pin_to_pin',
-    partitioned_by = ARRAY['year'],
-    bucketed_by = ARRAY['pin10'],
-    bucket_count = 1
-) AS (
-    SELECT
-        pin10,
-        nearest_neighbor_1_pin10,
-        nearest_neighbor_1_dist_ft,
-        nearest_neighbor_2_pin10,
-        nearest_neighbor_2_dist_ft,
-        nearest_neighbor_3_pin10,
-        nearest_neighbor_3_dist_ft,
-        year
-    FROM proximity.dist_pin_to_pin_temp
-    UNION
-    SELECT
-        pin10,
-        nearest_neighbor_1_pin10,
-        nearest_neighbor_1_dist_ft,
-        nearest_neighbor_2_pin10,
-        nearest_neighbor_2_dist_ft,
-        nearest_neighbor_3_pin10,
-        nearest_neighbor_3_dist_ft,
-        year
-    FROM proximity.dist_pin_to_pin_temp2
-);
-
-DROP TABLE IF EXISTS proximity.dist_pin_to_pin_temp
-DROP TABLE IF EXISTS proximity.dist_pin_to_pin_temp2

```

Comment:
It didn't seem like this temporary table approach was necessary anymore; I couldn't figure out what it was adding, especially because there didn't seem to be any definition for `proximity.dist_pin_to_pin_temp2` anymore. Does that sound right to you?

Reply:
Ah, a bit of complexity/stupidity here. This query running successfully is highly dependent on the size of the buffer defined here:

```sql
INNER JOIN pin_locations AS loc
    ON ST_CONTAINS(
        ST_BUFFER(ST_POINT(dp.x_3435, dp.y_3435), 1000), loc.point
    )
```

The smaller the buffer, the more quickly it runs, but the less likely it is to find distances for all PINs.

Basically, this query is saying "find all PIN distances within 1km of the target PIN centroid". BUT, not all PINs in Cook actually have neighbors within 1km. Expanding the buffer to 10km will cover all PINs, but makes the query take ages to run (and usually fail).

So, my (undocumented) solution was to run the query twice, first using a 1km buffer, then expanding the buffer to 10km _only for PINs which did not have 3 neighbors within the 1km buffer_.

**suggestion:** This might actually be the perfect use-case for dbt, as we could simply define that series of transforms in the DAG. We can create a 1km buffer materialized table, a 10km buffer table, then a simple view that combines them.

Alternatively, it's possible that Athena can now just run this query using the 10km buffer. When we switched to V3 query engine it did speed up some of our other queries, maybe this one will get a boost too. Regardless, I recommend we open a separate issue to deal with this.

Reply:
Interesting, thanks for the insight @dfsnow! I also like your proposed solution of using a series of intermediate tables. Perhaps we open a follow-up story to implement that solution, and in the meantime remove this model from the DAG and continue treating it as a source for now?

Reply:
@jeancochrane I like that fix. Let's go ahead and do it!

Reply:
Great, `proximity.dist_pin_to_pin` has been reverted to a source in f2afd7d4aa03e034abe203b5fdb1f7a7d0390f09 and I opened https://github.com/ccao-data/data-architecture/issues/122 to track the work of adding it to the DAG!